frontend/src/scenes/dashboard/DashboardTemplateVariablesLogic.ts

```diff
@@ -0,0 +1,35 @@
+import { actions, kea, path, reducers } from 'kea'
+import { DashboardTemplateVariableType, FilterType, Optional } from '~/types'
+
+import type { dashboardTemplateVariablesLogicType } from './DashboardTemplateVariablesLogicType'
+
+export const dashboardTemplateVariablesLogic = kea<dashboardTemplateVariablesLogicType>([
+    path(['scenes', 'dashboard', 'DashboardTemplateVariablesLogic']),
+    actions({
+        setVariables: (variables: DashboardTemplateVariableType[]) => ({ variables }),
+        setVariable: (variable_name: string, filterGroup: Optional<FilterType, 'type'>) => ({
+            variable_name,
+            filterGroup,
+        }),
+    }),
+    reducers({
+        variables: [
+            [] as DashboardTemplateVariableType[],
+            {
+                setVariables: (_, { variables }) => variables,
+                setVariable: (
+                    variables: DashboardTemplateVariableType[],
+                    { variable_name, filterGroup }
+                ): DashboardTemplateVariableType[] => {
+                    // TODO: handle actions as well as events
+                    return variables.map((v: DashboardTemplateVariableType) => {
+                        if (v.name === variable_name && filterGroup?.events?.length && filterGroup.events[0]) {
+                            v.default = filterGroup.events[0]
+                        }
+                        return v
+                    })
```

Comment:
In redux/kea, you're not supposed to mutate variables in reducers. Here you're making a shallow copy of the existing array, but then mutating the objects that are in there.

Line 27 should be `return { ...v, default: filterGroup.events[0] }`.

Also, it's usually a convention to call the existing value in a reducer `state`. You don't need to put the type on it either, it should be autodetected.